packages/cloud/src/cloud/types.ts

```diff
@@ -0,0 +1,94 @@
+import { Client, ClientOptions } from '@portive/client';
+import { PlateEditor, Value } from '@udecode/plate-core';
+import { ReactEditor } from 'slate-react';
+import { Upload } from '../upload';
+import { createUploadStore } from '../upload/createUploadStore';
+
+/**
+ * Specifies just the `options` part of the CloudPlugin
+ */
+export type CloudPlugin = ClientOptions & {
+  uploadStoreInitialValue?: Record<string, Upload>;
+};
+
+export type CloudEditor<V extends Value = Value> = PlateEditor<V> &
+  ReactEditor &
```

Comment:
```suggestion

```
already included in `PlateEditor`

Reply:
Hi Ziad, this one breaks the types when I remove `ReactEditor`. PlateEditor appears to extend not `ReactEditor` but a `TReactEditor` that is a subset of ReactEditor. When I use Transforms.select against the `editor` (when it is `TReactEditor`) then it gives a type error.

Not sure what I should do here. I could typecast the `editor` which is `TReactEditor` to `ReactEditor` but it feels like it defeats the purpose. Or I could create a custom `Transforms.select` but that feels odd.

Let me know what you'd prefer.

Reply:
Fixed